pkgs/applications/office/gnucash/default.nix

```
@@ -1,106 +1,210 @@
-{ fetchurl, lib, stdenv, pkg-config, makeWrapper, cmake, gtest
-, boost, icu, libxml2, libxslt, gettext, swig, isocodes, gtk3, glibcLocales
-, webkitgtk, dconf, hicolor-icon-theme, libofx, aqbanking, gwenhywfar, libdbi
-, libdbiDrivers, guile, perl, perlPackages
+{
+  # nix "native" dependencies
+  fetchurl
+, lib
+, stdenv
+
+, glibcLocales  # necessary to pass locale-related GnuCash tests
+, wrapGAppsHook # nix sugar for Gnome apps
+
+, makeWrapper   # used to wrap gnucash with Gnome sugar and the cli utils which need Perl
+
+, cmake         # tell nix to use cmake rather than make
+
+  # GnuCash requirements
+, aqbanking
+, boost
+, glib
+, gtest
+, guile
+, gwenhywfar
+, icu
+, libdbi
+, libdbiDrivers
+, libofx
+, libxml2
+, libxslt
+, perl
+, pkg-config
+, swig
+, webkitgtk
+
+  # GnuCash cli utils' Perl requirements
+, perlPackages
 }:
 
-let
-
-  # Enable gnc-fq-* to run in command line.
-  perlWrapper = stdenv.mkDerivation {
-    name = perl.name + "-wrapper-for-gnucash";
-    nativeBuildInputs = [ makeWrapper ];
-    buildInputs = [ perl ] ++ (with perlPackages; [ FinanceQuote DateManip ]);
-    dontUnpack = true;
-
-    installPhase = ''
-      mkdir -p $out/bin
-      for script in ${perl}/bin/*; do
-        makeWrapper $script $out''${script#${perl}} \
-          --prefix "PERL5LIB" ":" "$PERL5LIB"
-      done
-    '';
-  };
-
-in
-
 stdenv.mkDerivation rec {
-  pname = "gnucash";
+  pname   = "gnucash";
   version = "4.9";
 
   src = fetchurl {
-    url = "mirror://sourceforge/gnucash/${pname}-${version}.tar.bz2";
-    sha256 = "sha256-mlUcMMG3EhmfwiJ6EJr7mE177xjhOBcLvHIlxsH6ty0=";
+    sha256 = "0bdpzb0wc9bjph5iff7133ppnkcqzfd10yi2qagij4mpq4q1qmcs";
+    url    = "https://github.com/Gnucash/gnucash/releases/download/${version}/gnucash-${version}.tar.bz2";
```

```suggestion
    url = "https://github.com/Gnucash/gnucash/releases/download/${version}/gnucash-${version}.tar.bz2";
    sha256 = "0bdpzb0wc9bjph5iff7133ppnkcqzfd10yi2qagij4mpq4q1qmcs";
```
The old sorting is the defacto standard. Also it is always a good idea to not unnecessarily increase the diff size.

---

fixed in 072900f31f1598aca35feb66e4c553bbe69a9509